test/e2e/proxy_test.go

```diff
@@ -68,7 +70,400 @@ func skipIfProxyDisabled(t *testing.T) {
+	skipIfKubeProxyEnabled(t, data)
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Local should be able to be connected from Node"
+	probeHostnameFromNode(t, nodeName(0), url, hostname0, errMsg)
+	probeHostnameFromNode(t, nodeName(1), url, hostname1, errMsg)
+}
+
+func testLoadBalancerLocalFromPod(t *testing.T, data *TestData, url, client0, client1, client0IP, client1IP, hostname0, hostname1 string) {
+	errMsg := "Server NodePort whose externalTrafficPolicy is Local should be able to be connected from Pod "
+	probeHostnameFromPod(t, data, client0, url, hostname0, errMsg)
+	probeHostnameFromPod(t, data, client1, url, hostname1, errMsg)
+	probeClientIPFromPod(t, data, client0, url, client0IP, errMsg)
+	probeClientIPFromPod(t, data, client1, url, client1IP, errMsg)
+}
+
+func TestProxyNodePortServiceIPv4(t *testing.T) {
+	skipIfNotIPv4Cluster(t)
+	testProxyNodePortService(t, false)
+}
+
+func TestProxyNodePortServiceIPv6(t *testing.T) {
+	skipIfNotIPv6Cluster(t)
+	testProxyNodePortService(t, true)
+}
+
+func testProxyNodePortService(t *testing.T, isIPv6 bool) {
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 2)
+	skipIfProxyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+
+	agnhost0 := "agnhost-0"
+	agnhost1 := "agnhost-1"
+	hostAgnhost0 := "agnhost-host-0"
+	hostAnghost1 := "agnhost-host-1"
+	busybox0 := "busybox-0"
+	busybox1 := "busybox-1"
+	node0 := nodeName(0)
+	node1 := nodeName(1)
+	localhostIP := "127.0.0.1"
+	busybox0Ipv4, busybox0Ipv6 := createTestClientPod(t, data, busybox0, node0)
+	busybox1Ipv4, busybox1Ipv6 := createTestClientPod(t, data, busybox1, node1)
+	busybox0IP := busybox0Ipv4
+	busybox1IP := busybox1Ipv4
+	node0IP := controlPlaneNodeIPv4()
+	node1IP := workerNodeIPv4(1)
+	if isIPv6 {
+		busybox0IP = busybox0Ipv6
+		busybox1IP = busybox1Ipv6
+		node0IP = controlPlaneNodeIPv6()
+		node1IP = workerNodeIPv6(1)
+	}
+	portStrCluster := createNodePortServices(t, data, "agnhost-cluster", false, isIPv6)
+	portStrLocal := createNodePortServices(t, data, "agnhost-local", true, isIPv6)
+	createAgnhostPod(t, data, agnhost0, node0, false)
+	createAgnhostPod(t, data, agnhost1, node1, false)
+	t.Run("Pod CIDR Endpoints", func(t *testing.T) {
+		nodePortTestCases(t, data, portStrCluster, portStrLocal, node0IP, node1IP, localhostIP, busybox0, busybox1, busybox0IP, busybox1IP, agnhost0, agnhost1, false)
+	})
+
+	require.NoError(t, data.deletePod(testNamespace, agnhost0))
+	require.NoError(t, data.deletePod(testNamespace, agnhost1))
+	createAgnhostPod(t, data, hostAgnhost0, node0, true)
+	createAgnhostPod(t, data, hostAnghost1, node1, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		nodePortTestCases(t, data, portStrCluster, portStrLocal, node0IP, node1IP, localhostIP, busybox0, busybox1, busybox0IP, busybox1IP, node0, node1, true)
```

Comment:
ditto, this kind of call is weird and error-prone.

Reply:
I meant the doubled number of parameters

Reply:
I have removed all doubled number of parameters.